documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1015,13 +1016,13 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
 
-### <a name="R2064" />R2064 LicenseMissing
-**Output Message**: Please provide correct licensing information here. Acceptable value: "name": "MICROSOFT\_MIT\_NO\_VERSION"
+### <a name="R2065" />R2065 LicenseHeaderMustNotBeSpecified
+**Output Message**: License header must not be specified in the OpenAPI document.
 
-**Description**: Each OpenAPI json document must contain the license object in its `info` section which must set the name to `MICROSOFT\_MIT\_NO\_VERSION`. This is necessary for AutoRest to insert the appropriate headers in the generated code.
+**Description**: `x-ms-code-generation-settings` must not have the header section specified in the OpenAPI documents since each SDK has a different licensing header, this information must be provided either from the command line or the configuration file.
```

Comment:
Original: having `info.license` is cool and all, but it is not used for code gen which this rule sounds like it was for: the `MICROSOFT_MIT_NO_VERSION` constant has no meaning in `info.license`, but in `info.x-ms-code-generation-settings.header` - this leads to the headers that go into all generated code. Now here is the situation: Different SDKs have different demands regarding that, so it really makes zero sense to put something that is not only codegen specific, but even language specific into a Swagger file. I think *most* SDKs ship with `MICROSOFT_MIT_NO_VERSION`, but not all (at least Python I think). In the past, we had people put `MIT` into that field which has no meaning, so in the end *all* SDK teams had fun editing the Swagger before being able to generate code.

TL;DR: this field was a bad idea, causes confusion and pain, doesn't belong in Swagger